src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -244,15 +292,21 @@ public SimpleArgument(IParameterSymbol parameter, IExpression value)
                 this.Parameter = parameter;
             }
 
-            public ArgumentKind Kind => ArgumentKind.Positional;
+            public ArgumentKind ArgumentKind => ArgumentKind.Positional;
 
             public IParameterSymbol Parameter { get; }
 
             public IExpression Value { get; }
 
-            public IExpression InConversion => null;
+            IExpression IArgument.InConversion => null;
+
+            IExpression IArgument.OutConversion => null;
+
+            bool IOperation.IsInvalid => this.Parameter == null || this.Value.IsInvalid;
+
+            OperationKind IOperation.Kind => OperationKind.Argument;
 
-            public IExpression OutConversion => null;
+            SyntaxNode IOperation.Syntax => this.Value.Syntax;
         }
 
         class Argument : IArgument
```

Comment:
It looks like SimpleArgument and Argument duplicate a lot of logic while implementing IArgument. Would it make sense to introduce a base class for them and move common code there?